File: PythonCode/1045.py
a,b,c = map(float, input().split())
lista = [a,b,c]
lista.sort()
lista.reverse()
if(lista[0]>=(lista[1]+lista[2])):
    print("NAO FORMA TRIANGULO")
elif(lista[0]**2)== ((lista[1]**2) + (lista[2]**2)):
    print("TRIANGULO RETANGULO")
elif(lista[0]**2) > ((lista[1]**2)+ (lista[2]**2)):
    print("TRIANGULO OBTUSANGULO")
elif(lista[0]**2) < ((lista[1]**2)+(lista[2]**2)):
    print("TRIANGULO ACUTANGULO")
if(lista[0] == lista[1] == lista[2]):
    print("TRIANGULO EQUILATERO")
if(lista[0] == lista[1] != lista[2]) or (lista[0] == lista[2] != lista[1]) or (lista[1] == lista[2] != lista[0]):
    print("TRIANGULO ISOSCELES")

File: C++Code/1045.cpp
//Questão: Tipos de triangulos
#include <iostream>
#include <vector>
#include <algorithm>
#include <math.h>
using namespace std;

int main()
{
    float x;
    vector<float> v;

    for (int i = 0; i < 3; ++i)
    {
        cin >> x;
        v.push_back(x);
    }

    sort(v.begin(), v.end(), greater<float>());

    if (v[0] >= v[1]+v[2])
    {
        cout << "NAO FORMA TRIANGULO" <<endl;
    }
    else if (pow(v[0],2) == pow(v[1],2) + pow(v[2],2))
    {
         cout << "TRIANGULO RETANGULO"<<endl;
    }
    else if (pow(v[0],2) > pow(v[1],2) + pow(v[2],2))
    {
         cout << "TRIANGULO OBTUSANGULO"<<endl;
    }
    else if (pow(v[0],2) < pow(v[1],2) + pow(v[2],2))
    {
         cout << "TRIANGULO ACUTANGULO"<<endl;
    }
    if ((v[0] == v[1]) && (v[0] == v[2]))
    {
         cout << "TRIANGULO EQUILATERO"<<endl;
    }
    if ( (v[0] == v[1] && v[0] != v[2]) || (v[0] == v[2] && v[0] != v[1]) || (v[1] == v[2] && v[1] != v[0]))
    {
       cout << "TRIANGULO ISOSCELES"<<endl;
    }

    return 0;
}